### ISSUE TITLE
Accuracy drops during extended training.

### ISSUE THREAD

**Issue by Sahel13** [Orkis-Research/Pytorch-Quaternion-Neural-Networks#13]
Hi,

I've built the following quaternion CNN using the methods provided.
```
class LeNet_300_100(nn.Module):
    """
    LeNet_300_100 neural network architecture.
    """
    def __init__(self):
        super().__init__()
        self.fc1 = QuaternionLinear(784, 300)
        self.fc2 = QuaternionLinear(300, 100)
        self.fc3 = nn.Linear(100, 10)

    def forward(self, x):
        x = torch.flatten(x, 1)
        x = F.relu(self.fc1(x))
        x = F.relu(self.fc2(x))
        x = self.fc3(x)
        return x
```
When training the model for an extended duration on the MNIST dataset, the accuracy suddenly drops to nearly 10%, which is what we would expect from an untrained model, and doesn't improve any further. An image of the accuracy values as training progresses is attached.

![accuracy](https://user-images.githubusercontent.com/18627035/137481946-5478aed2-c62d-4858-b5d6-d96ea6dff04d.png)

I would appreciate some insight into why this might be happening. If you need additional info, I can provide the code to recreate this issue.

Thanks,
Sahel
